### PR TITLE
More intuitive API to configure transaction levels

### PR DESCRIPTION
**Pull request by andreasohlund**
This pull introduces a better way of configuring transactions. Instead of `.Transactions().Enable|Disable|EnableDistributedTransactions|DisableDistributedTransactions` we now have a single

```
config.UseTransport<T>()
    .Transactions(TransportTransactionMode.None|ReadOnly|SendsAtomicWithReceive|Promotable)

```

Not specifying a mode will cause the core to use the highest one supported by the transport. An exception will be thrown if a non supported mode is requested.

This is aligns better with what the transports support and also allows multi queue transactions to be requested (supported by SQL and MSMQ and to some extent ASB)

Hopefully this will cause less confusion for the users.

### Questions:

* Are we happy with `.Transactions()`?
* Is `TransportTransactionMode.X` a good name?

Doco pull: https://github.com/Particular/docs.particular.net/pull/979

### CODE REVIEW

**Comment by andreasohlund** [comment]
@Particular/nservicebus still WIP but I would appreciate a review to get feedback on the naming.

**Comment by timbussmann** [comment]
TransactionMode seems reasonable to me. TransactionScope would also fit quite well, but sadly that's no real option.

Any reason why the Enum has a different name than the configuration option? e.g. why not `TransactionMode(TransactionMode.SingleQueue)`

**Comment by andreasohlund** [comment]
>why not TransactionMode(TransactionMode.SingleQueue)

No reason, I'll rename

**Comment by danielmarbach** [comment]
Or just `Transactions(TransactionMode.SingleQueue)`

**Comment by andreasohlund** [comment]
Renames pushed, went with @danielmarbach proposal

**Comment by andreasohlund** [comment]
Updated names after call with @SzymonPobiega @danielmarbach and @timbussmann

File: src/NServiceBus.Core/Transports/TransportDefinition.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public abstract partial class TransportDefinition
         /// <summary>
         /// Gets the supported transactionality for this transport.
         /// </summary>
-        public abstract TransactionSupport GetTransactionSupport();
+        public abstract TransportTransactionMode GetTransactionSupport();
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Transports/TransportDefinition.cs]: `GetTransactionMode`?

Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/TransportDefinition.cs]: How about `GetSupportedTransactionMode` ?

Comment (danielmarbach) [inline src/NServiceBus.Core/Transports/TransportDefinition.cs]: :+1:

Comment (andreasohlund) [inline src/NServiceBus.Core/Transports/TransportDefinition.cs]: Done

**Comment by andreasohlund** [comment]
Doco pull is here https://github.com/Particular/docs.particular.net/pull/979

@SzymonPobiega @danielmarbach this one is ready to go

**Comment by yvesgoeleven** [comment]
What if we remove the word 'Transaction' and instead focus on the capabilities that we're looking for, like receive rollback, retry, send rollback, storage operation rollback etc... some transports (ASQ & ASB) offer these capabilities, but they don't use transactions

**Comment by andreasohlund** [comment]
I guess you could argue that the peek locks are a form or transaction implementation but I see what you're getting at. ASB seems to use the term "receive mode" - https://msdn.microsoft.com/en-us/library/microsoft.servicebus.messaging.receivemode.aspx so perhaps that is better.

That said I wonder if users are still more comfortable with the word "transaction". What do you think @Particular/nservicebus ?

**Pull request merged**